### PR TITLE
Add support for DS Proxy

### LINKED ISSUE #27: Add support for DS Proxy
We should add DS Proxy support to the DS Operator.

### PR DESCRIPTION
**Pull request by jrcast**
1. Add support for DS Proxy
1. Add to support to change DS' `groupID`
1. Add deployment access to the `ds-operator`'s RBAC
1. Set `ImagePullPolicy` to `PullIfNotPresent`
1. Remove finalizers. This functionality is not in use yet
1. Fix linter warnings
    
Closes #27

### CODE REVIEW

File: controllers/proxy.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
+/*
+	Copyright 2020 ForgeRock AS.
```
Comment (wstrange) [inline controllers/proxy.go:2]: 2021

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -28,7 +28,7 @@ spec:
         args:
         - --enable-leader-election
         image: controller:latest
-        imagePullPolicy: Always
+        imagePullPolicy: IfNotPresent
```
Comment (jrcast) [inline config/manager/manager.yaml:31]: As per https://kubernetes.io/docs/concepts/containers/images/#updating-images , you can force image pulls by leaving the default setting: `IfNotPresent` + using the tag `latest`. `Always` is not required.

**Review by wstrange — APPROVED** [review]
super nice Jhonattan.

File: controllers/backup.go [hunk 3]
```diff
@@ -17,8 +18,13 @@ func (r *DirectoryServiceReconciler) updateBackup(ctx context.Context, ds *direc
 
 	if !ds.Spec.Backup.Enabled {
 		log.V(5).Info("Backup is disabled. Nothing to do")
-		// Delete the backup tasks - even if they dont exist - this is no more expensive than querying then deleting..
-		l.DeleteBackupTask(ds.Name)
+		bStatus, err := l.GetBackupTaskStatus(ds.Name)
```
Comment (jrcast) [inline controllers/backup.go:21]: This reduces `"FAILED"` error messages in DS logs. We were trying to delete the Tasks without checking if they were there.

File: controllers/sts.go [hunk 4]
```diff
@@ -142,7 +143,7 @@ func createDSStatefulSet(ds *directoryv1alpha1.DirectoryService, sts *apps.State
 						{
 							Name:            "init",
 							Image:           ds.Spec.Image,
-							ImagePullPolicy: v1.PullAlways, // todo: for testing this is good. Remove later?
+							ImagePullPolicy: v1.PullIfNotPresent,
```
Comment (jrcast) [inline controllers/sts.go:146]: As per https://kubernetes.io/docs/concepts/containers/images/#updating-images , you can force image pulls by leaving the default setting: `IfNotPresent` + using the tag `latest`. `Always` is not required.

**Pull request merged**